### ISSUE TITLE
V2 issue: AppConfig should not have v2 future flags

### ISSUE THREAD

**Issue by kentcdodds** [remix-run/remix#7253]
### What version of Remix are you using?

0.0.0-nightly-49e8da1-20230823

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

[TS Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzjAnmApnAgmMBhCAOwDNgBzOAXzmKghDgCIABKdEYADwFooBXQgHoAJugBujANxxBg5GnQBnAFxwADADpNa7oXIALGABsU3ACwBOdAA5hAQwCM3AExrnAZjXWPAKB-pOSFg4UWI7PiN4BB8ASABjOzj9dAARYDY4mGgUVQByDUFCCFEAfRBiiKVBDQSk9EE2Dk5cgBpY8iK2YQAlCD4YdAAxYCMlVQBtXIAqKcEp3IBdNpjFdCgxNYBZCtHB6BA7GDylEFbY1fW1gAUjQ+J9vKLRM5iYOxGAd2BCYVUYfnQy0gihgcUUKmQAOWH0OSSuh30EMmBTen2+whqRFIZA0MEUi2WxH6fDYqmicApMjkMGSqzgin0fSMwjgRXgACNMHYjEYIB90L9YjExM4Ssk7KIoBD-nxAUKRWV0G8-lD5aK1nQoAAhPo-OxQHKQ2XLYWizoHIzAABeQ32myVjN+RrlMVNJTo-XQBEIG0IMGARBVxrVJVEYiDLsobWoikOwEUpCU2Fw3uxPiAA):

```tsx
import { type AppConfig } from "@remix-run/dev"; // types: 0.0.0-nightly-49e8da1-20230823

export default {
	cacheDirectory: './node_modules/.cache/remix',
	ignoredRouteFiles: ['**/*'],
	serverModuleFormat: 'esm',
	serverPlatform: 'node',
	tailwind: true,
	postcss: true,
	watchPaths: ['./tailwind.config.ts'],
	future: {
    // these should not be allowed:
		v2_headers: true,
		v2_meta: true,
		v2_errorBoundary: true,
		v2_normalizeFormMethod: true,
		v2_routeConvention: true,
		v2_dev: true,
	},
} satisfies AppConfig
```

### Expected Behavior

v2 future flags should not be allowed with nightly (v2)

### Actual Behavior

v2 future flags are allowed in nightly (v2)

**Comment by brophdawg11** [comment]
Huh, interesting.  That type is `Partial<FutureConfig>` and `FutureConfig` is empty, but I guess TS allows other props since they satisfy the base value (https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks).

I guess we can union that with a `never`: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIFczqhAwgexBmAHNkBvAKAEgA3AZgH0o4ATVCOLHALmQCN8+ADacQlAL6VKoSLEQoAggAdlBIqQqVkO5DEzYIAfj4AFOFDDA4wgDwZueQsRIA+ZADIt1agG1lUPjKAHJwALYQfADOYFCgJAC6fCAQtNAA3DQSmVKUEAAeyviWyKwQ8OjCYN76jnxUPgzMbBxchnyx6BAANDR0AEyMAO5gABYQCADWHVBdvdQSvRLIUVzAUcQQUcgqas6klEA

**Comment by kentcdodds** [comment]
That sounds good. I'm just sure many people will upgrade and they'll have to run the server to realize they still need to remove the old feature flags, so this is probably worth it.

**Comment by brophdawg11** [comment]
Yeah - since the config is not TS by default we did add some build time warnings for those in https://github.com/remix-run/remix/blob/dev/packages/remix-dev/config.ts#L566.  But if folks are using TS like this it would be nice to let them know 👍

**Comment by kentcdodds** [comment]
Agreed, those warnings are definitely nice 👍 Thanks!

**Comment by brophdawg11** [comment]
Fixed by #7261 and will be included in Remix v2
